src/libPersistence/ContractStorage.cpp

```diff
@@ -403,7 +403,7 @@ bool ContractStorage::FetchStateValue(const dev::h160& addr,
 bool ContractStorage::FetchExternalStateValue(
     const dev::h160& /*caller*/, const dev::h160& target, const zbytes& src,
     unsigned int s_offset, zbytes& dst, unsigned int d_offset, bool& foundVal,
-    string& type, uint32_t caller_version) {
+    string& type, uint32_t caller_version, Account* injected) {
```

Comment:
add the option to 'inject' an account when getting the state value (since there is a lot of protobuf logic in here)